auto_run.py

```diff
@@ -0,0 +1,148 @@
+#%% 
+import hourglass
+import imp
+imp.reload(hourglass)
+from hourglass import HourglassNet
+from constants import *
+import matplotlib.pyplot as plt
+import os
+
+import data_generator
+imp.reload(data_generator)
+import data_generator
+import time
+
+import re
+import pandas as pd
+import evaluation
+from evaluation import Evaluation
+imp.reload(evaluation)
+from HeatMap import HeatMap
+
+#import HeatMap
+#imp.reload(HeatMap)
+
+import numpy as np
+
+#%% find duplicates in list
+
+def find_dup(pair_list, element):
+    for (a,b) in pair_list:
+        if element == a or element == b:
+            return True
+    return False
+
+#%% 
+
+def find_epochs(base_dir, sub_dir, epoch_dic, visited_models):
+    directory = os.listdir(os.path.join(base_dir, sub_dir))
+    weight_file = [f for f in directory if (f.endswith(".hdf5"))]
+    for name in weight_file:
+        match = re.search('hpe_epoch(\d+)', name)
+        if match:
+            epoch.append((match.group(1), sub_dir))
```

Comment:
Recommend dictionary here
```suggestion
    epochs = {}
    directory = os.listdir(os.path.join(base_dir, sub_dir))
    weight_file = [f for f in directory if (f.endswith(".hdf5"))]
    for name in weight_file:
        match = re.search('^hpe_epoch(\d+)', name)
        if match:
            epochs[match.group] = sub_dir
```

Reply:
Oh yeah good suggestion :)